src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs

```diff
@@ -140,42 +138,49 @@ public async Task CommitFilesAsync(List<GitFile> filesToCommit, string repoUri,
     ///     - If no setting, or if auto, then determine whether incoming content differs in line ends vs. the
     ///       OS setting, and replace if needed.
     /// </remarks>
-    private string NormalizeLineEndings(string filePath, string content)
+    private async Task<string> NormalizeLineEndings(string repoPath, string filePath, string content)
     {
         const string crlf = "\r\n";
         const string lf = "\n";
+
         // Check gitAttributes to determine whether the file has eof handling set.
-        string eofAttr = LocalHelpers.ExecuteCommand(_gitExecutable, $"check-attr eol -- {filePath}", _logger);
-        if (string.IsNullOrEmpty(eofAttr) ||
-            eofAttr.Contains("eol: unspecified") ||
-            eofAttr.Contains("eol: auto"))
-        {
-            if (Environment.NewLine != crlf)
-            {
-                return content.Replace(crlf, Environment.NewLine);
-            }
-            else if (Environment.NewLine == crlf && !content.Contains(crlf))
-            {
-                return content.Replace(lf, Environment.NewLine);
-            }
-        }
-        else if (eofAttr.Contains("eol: crlf"))
-        {
-            // Test to avoid adding extra \r.
-            if (!content.Contains(crlf))
-            {
-                return content.Replace(lf, crlf);
-            }
-        }
-        else if (eofAttr.Contains("eol: lf"))
-        {
-            return content.Replace(crlf, lf);
-        }
-        else
+        var result = await _processManager.ExecuteGit(repoPath, new[] { "check-attr", "eol", "--", filePath });
+        result.ThrowIfFailed($"Failed to determine eol for {filePath}");
+
+        string eofAttr = result.StandardOutput.Trim();
+
+        switch (eofAttr)
```

Comment:
I think the original if/else statement was clearer. Aside from IsNullOrEmpty, all of these comparisons are Contains, which seems awkward in a switch statement (not comparisons against a constant). Also, I think the `null` case is duplicated anyway.

Reply:
Reverted